libs/wingsdk/src/target-sim/counter.inflight.ts

```diff
@@ -25,6 +27,13 @@ export class Counter implements ICounterClient, ISimulatorResourceInstance {
 
   public async cleanup(): Promise<void> {}
 
+  public async save(dir: string): Promise<void> {
```

Comment:
- instead of in memory, can we just store the values all the time into the state directory and have that be the source of truth?

Much more robust